src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -899,8 +894,7 @@ void WindowImplX11::setIcon(const Vector2u& size, const std::uint8_t* pixels)
 {
     // X11 wants BGRA pixels: swap red and blue channels
     // Note: this memory will be freed by XDestroyImage
-    auto* iconPixels = static_cast<std::uint8_t*>(
-        std::malloc(static_cast<std::size_t>(size.x) * static_cast<std::size_t>(size.y) * 4));
+    auto iconPixels = std::vector<std::uint8_t>(size.x * size.y * 4);
```

Comment:
Notice the comment above. This is supposed to be allocated with malloc and freed in XDestroyImage.
Maybe our XPointerDeleter operator() could have an overload for XImage* that calls XDestroyImage?

Reply:
I was annoyed that `XDestroyImage` forced us to use `malloc` to create that buffer so by not using `XDestroyImage` we also get to use C++ idioms for allocating and freeing that buffer while `XFree` continues to clean up the image.

Here is the [`XDestroyImage`](https://github.com/mirror/libX11/blob/ff8706a5eae25b8bafce300527079f68a201d27f/src/Macros.c#L245-L249) implementation for reference. It doesn't appear to be doing anything magical but maybe I'm missing something.

Reply:
Having to look at a dependency's the source code (and actually not looking at [it](https://github.com/mirror/libX11/blob/ff8706a5eae25b8bafce300527079f68a201d27f/src/ImUtil.c#L435-L441)) instead of using the documented public interface seems wrong. It goes past libX's abstraction barrier, increasing coupling with a specific implementation. Using malloc may feel annoying, but I think it is the correct thing to do here.